### PR TITLE
JAX Implementation of Power Spherical logpdf & pdf

### PR DESCRIPTION
**Pull request by harryjulian**
As referenced in #11737. 

Paper is [here.](https://arxiv.org/pdf/2006.04437.pdf)

Initial implementation. Tested it in notebooks re: the authors [example](https://github.com/nicola-decao/power_spherical) and the [Tensorflow Probability implementation](https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/PowerSpherical) and looks good. 

This one is a little less straightforward; I've got a few of questions:

1) As these functions aren't in `scipy.stats`, should they live there? It makes the most sense to be with all other distributions imo.
2) When it comes to writing tests, as these functions aren't in `scipy.stats`, how should I go about validating their correctness without adding extra dependencies?
3) As the two arguments `mu` and `kappa` are usually of alternative shapes, what is the most appropriate way to conduct type promotion?

Edit: looks like I made a bit of a mess with this PR, I'll re-open post discussion.

### CODE REVIEW

**Comment by jakevdp** [comment]
Hi - thanks for the contribution. I think this lies in a bit of a gray area because the corresponding distribution isn't implemented in `scipy.stats`. We don't have any hard policy regarding what is in scope for `jax.scipy.stats`, but generally adding functionality beyond what is in scipy requires some discussion.

Who do you anticipate is the audience for this functionality? Can we think of a better home for it rather than including it in jax itself?

**Comment by harryjulian** [comment]
Figures. Taking a quick look at the [citations](https://scholar.google.com/scholar?cites=13182117127367358760&as_sdt=2005&sciodt=0,5&hl=en) the paper has on scholar, it looks like it's mainly used in a variety of more niche ML use cases (robotics, an embedded loss function, generative modelling applications). As the work is quite new, perhaps its main use cases are yet to come?

I can't think of any other specific Jax-based libraries which would be a better fit, off the top of my head.

**Comment by jakevdp** [comment]
I think I'm going to close this – it's not clear that these routines are in scope for `jax.scipy`. Thanks for the contribution, and sorry we weren't able to merge it this time!
